Various fixes for the WebDAV-API

open-dynaMIX opened the pull request:
- fix: only serve newest File form Document
  This is needed for multiple reasons:

   1. There is no need to edit old files and the result would probably be
      unexpected
   2. When creating a new version, Libreoffice doesn't know about this and
      it would use the older File as startingpoint. Leading to a new
      version on every save.
 - fix: only create webDAV links for original Files
 - fix: run hooks after file save was successful
 - chore: simplify fetching user attributes from request

winged commented on alexandria/core/serializers.py:
```diff
@@ -15,7 +15,11 @@
 
 def default_user_attribute(user, attribute):
-    return getattr(user, attribute) if not isinstance(user, AnonymousUser) else None
+    return (
+        getattr(user, attribute)
+        if user is not None and not isinstance(user, AnonymousUser)
+        else None
+    )
```
Could we do this instead?

```suggestion
    return getattr(user, attribute, None)
```

winged replied:
This will also just return `None` if you mistype an attribute on a regular user object, so maybe not the best idea, but we'd fail hard in that case anyway

open-dynaMIX replied:
I went a different path. I don't feel like there needs to be a function that does nothing more than `getattr`.

winged reviewed (approved):
LGTM, just a suggestion to simplify the `default_user_attribute()` thingy a bit (not 100% sure it fully covers our needs, but should be logically equivalent)

winged reviewed (commented):
Nice! But I think it would make sense to get rid of the `default_user_attribute()` function entirely (ideally in a separate PR or at least in it's own commit)

open-dynaMIX commented:
Agree. I've added a commit for that.

> Nice! But I think it would make sense to get rid of the `default_user_attribute()` function entirely (ideally in a separate PR or at least in it's own commit)

I agree. For keeping it simple, I've added a commit for that to this PR.
